wger/nutrition/models.py

```diff
@@ -137,10 +138,17 @@ def get_nutritional_values(self):
             for key in result['percent'].keys():
                 result['percent'][key] = \
                     result['total'][key] * ENERGY_FACTOR[key][unit] / energy * 100
-
+        
         # Per body weight
         if self.user.userprofile.weight:
-            weight = Decimal(self.user.userprofile.weight)
+            # Determine closest entry by date
+            target = self.creation_date
+            closest_entry = None
+            for entry in WeightEntry.objects.filter(user = self.user).order_by('date').iterator():
+                if not closest_entry or \
+                (abs(entry.date - self.creation_date) <= abs(closest_entry.date - self.creation_date)):
+                    closest_entry = entry
+            weight = Decimal(closest_entry.weight)
```

Comment:
Ironically, I actually had it that way originally.

The try-excepts should be unnecessary since we're using less than or equal to and have already confirmed one entry exists with the conditional on 143. Please correct me if you think my logic is off.